nemo/collections/asr/parts/utils/nmse_clustering.py

```diff
@@ -137,135 +123,263 @@ def getLaplacian(X):
     return L
 
 
-def eig_decompose(L, k):
-    try:
-        lambdas, eig_vecs = scipy.linalg.eigh(L)
-    except:
-        try:
-            lambdas = scipy.linalg.eigvals(L)
-            eig_vecs = None
-        except:
-            lambdas, eig_vecs = scipy.sparse.linalg.eigsh(L)  ### Inaccurate results
-    return lambdas, eig_vecs
+def eigDecomposeGPU(Laplacian):
+    device = torch.cuda.current_device()
+    laplacian_torch = torch.from_numpy(Laplacian).float().to(device)
+    lambdas_torch, _ = torch.linalg.eigh(laplacian_torch)
+    lambdas = lambdas_torch.cpu().numpy()
+    return lambdas
 
 
-def getLamdaGaplist(lambdas):
-    lambda_gap_list = []
-    for i in range(len(lambdas) - 1):
-        lambda_gap_list.append(float(lambdas[i + 1]) - float(lambdas[i]))
-    return lambda_gap_list
+def eigDecomposeCPU(Laplacian):
+    lambdas, _ = scipy.linalg.eigh(Laplacian)
```

Comment:
if its on cpu, can't we use torch linalg? if yes, we can just avoid dependency on scipy

Reply:
I removed scipy and replaced scipy eigh with torch version. Now we don't need scipy at all.